src/votekit/ballot_generator.py

```diff
@@ -0,0 +1,534 @@
+from abc import abstractmethod
+from fractions import Fraction
+from typing import Optional, Union
+from .profile import PreferenceProfile
+from .ballot import Ballot
+from numpy.random import choice
+import itertools as it
+import random
+import numpy as np
+import pickle
+from pathlib import Path
+from itertools import zip_longest
+
+"""
+IC
+IAC
+1d spatial
+2d spatial
+PL
+BT
+AC
+Cambridge
+"""
+
+
+class BallotGenerator:
+
+    # cand is a set
+    # number_of_ballots: int
+    # candidates: list
+    # ballot_length: Optional[int]
+    # slate_to_candidate: dict  # race: [candidate]
+    # pref_interval_by_slate: dict = None  # race: {candidate : interval length}
+    # hyperameters: dict cohesion, Direchlet alphas
+    # demo_breakdown: dict = None  # race: percentage
+
+    def __init__(
+        self,
+        number_of_ballots: int,
+        candidates: Union[list, dict],
+        ballot_length: Optional[int] = None,
+        hyperparameters: Optional[dict] = {},
+    ):
+        self.number_of_ballots = number_of_ballots
+        if isinstance(candidates, dict):
+            self.slate_to_candidate = candidates
+            self.candidates = list(
+                {cand for cands in candidates.values() for cand in cands}
+            )
+        else:
+            self.candidates = candidates
+        self.ballot_length = (
+            ballot_length if ballot_length is not None else len(self.candidates)
+        )
+
+        if hyperparameters:
+            if isinstance(candidates, dict):  # add type error
+                self.set_params(candidates, **hyperparameters)
+                self.parameterized = True
+            else:
+                raise TypeError(
+                    "'candidates' must be dictionary when hyperparameters are set"
+                )
+
+    @abstractmethod
+    def generate_profile(self) -> PreferenceProfile:
+        pass
+
+    # @staticmethod
+    # def list_to_set(candidates):
+    #     return [set(cand) for cand in candidates]
+
+    @staticmethod
+    def ballot_pool_to_profile(ballot_pool, candidates):
```

Comment:
just a note on "pythonic", especially since the rest of your code has excellent style overall. 

Most people don't use `staticmethod`, when a function works as a staticmethod (as opposed to `classmethod`, which is useful!) it takes no parameters relevant to the class (i.e. no `self`, no `cls`) and is best served as a module-level utility method.

That said, this is a 100% fine way to do things, but unless you're intentionally leaving the window open for a class to override it, I'd say 9/10 python codebases would just move this to the module-level instead of class-level.